swift/tests/InputSampleFiles/testfile.swift

```diff
@@ -576,4 +576,52 @@ class SwiftExamples {
        someObject.isEnabled = cachedExperiments.isTreated(forExperiment: xpName)
        return cachedExperiments.isTreated(forExperiment: xpName)
     }
+    
+    private func ifElseLadder_1() {
+        if x {
+            print("1")
+        } else if cachedExperiments.isTreated(forExperiment: ExperimentNamesSwift.test_experiment) {
+            print("2")
+        } else if cachedExperiments.isTreated(forExperiment: ExperimentNamesSwift.test_experiment1) {
+            print("3")
+        } else {
+            print("4")
+        }
+    }
+    
+    private func ifElseLadder_2() {
+        if x {
+            print("1")
+        } else if cachedExperiments.isTreated(forExperiment: ExperimentNamesSwift.test_experiment1) {
+            print("2")
+        } else if cachedExperiments.isTreated(forExperiment: ExperimentNamesSwift.test_experiment) {
+            print("3")
+        } else {
+            print("4")
+        }
+    }
+    
+    private func ifElseLadder_3() {
+        if x {
+            print("1")
+        } else if true {
+            print("2")
+        } else if cachedExperiments.isTreated(forExperiment: ExperimentNamesSwift.test_experiment) {
+            print("3")
+        } else {
+            print("4")
+        }
+    }
+    
+    private func ifElseLadder_4() {
```

Comment:
Can you also add couple of ladder tests that 
1. test compound expressions reducing to a boolean as part of this set of tests? 
2. test ending with a else if instead of else. 

e.g.,
```
if x {
   ...
} else if cachedExperiments.isTreated(...stale) || y  {
   ... 
} else if cachedExperiments.isTreated(...nonstale) || z { 
   ... 
} else {
   ...
}

Reply:
Yeah, added for both the cases
1. ifElseLadder_5 - having "or" expression.
2. ifElseLadder_6 - having "and" expression.
3. ifElseLadder_6 - if else terminating with "else-if"..